src/gc/gc.cpp

```diff
@@ -7389,6 +7389,13 @@ int gc_heap::grow_brick_card_tables (uint8_t* start,
 
         seg_mapping_table = new_seg_mapping_table;
 
+        // The above write of the card table base address, is read by executing threads via the instruction cache.
```

Comment:
I do not think that there is instruction cache involved here. `g_gc_card_table` is regular static, it won't be read via the instruction cache.

The comment about instruction cache should be next to https://github.com/dotnet/coreclr/blob/2dfc74df1f9fe238eb2ec25c4764b8b65432b010/src/vm/gcenv.ee.cpp#L1281 as @mjp41 pointed out. It is where we are patching the code inline.